monitor.py

```diff
@@ -367,109 +427,35 @@ def get_quota_and_nightly(spy_link):
-            nightly_log_re = re.compile('(Resolved release s390x-latest to (\S+))', re.MULTILINE|re.DOTALL)
-            nightly_log_match = nightly_log_re.search(build_log_response.text)
-            if nightly_log_match is None:
-                rc_nightly_log_re = re.compile('(Using explicitly provided pull-spec for release s390x-latest \((\S+)\))', re.MULTILINE|re.DOTALL)
-                rc_nightly_log_match = rc_nightly_log_re.search(build_log_response.text)
-                if rc_nightly_log_match is None:
-                    nightly = "Unable to fetch nightly information- No match found"
-                else:
-                    nightly = rc_nightly_log_match.group(2)
-            else:
-                nightly = "s390x-latest-"+ nightly_log_match.group(2)
-            return lease, nightly
-        except requests.Timeout:
-            return "Request timed out"
-        except requests.RequestException:
-            return "Error while sending request to url"
-    else:     
-        build_log_url = PROW_VIEW_URL + spy_link[8:] + "/build-log.txt"
-        try:
-            build_log_response = requests.get(build_log_url, verify=False, timeout=15)
-            # lease is not applicable for SNO hence checking only for MCE
-            if "mce" in spy_link:
-                job_platform = "aws" #currently it contains only aws hence hardcoding
-                zone_log_re = re.compile('(Acquired 1 lease\(s\) for {}-quota-slice: \[)([^]]+)(\])'.format(job_platform), re.MULTILINE|re.DOTALL)
-                zone_log_match = zone_log_re.search(build_log_response.text)
-                if zone_log_match is None:
-                    lease = "Failed to fetch lease information"
-                else:
-                    lease = zone_log_match.group(2)
-            nightly_log_re = re.compile('(Resolved release multi-latest to (\S+))', re.MULTILINE|re.DOTALL)
-            nightly_log_match = nightly_log_re.search(build_log_response.text)
-            if nightly_log_match is None:
-                nightly = "Failed to fetch nightly image"
-            else:
-                nightly = "multi-latest-"+ nightly_log_match.group(2)
-            return lease, nightly
-        except requests.Timeout:
-            return "Request timed out"
-        except requests.RequestException:
-            return "Error while sending request to url"
-
+    build_log_url = PROW_VIEW_URL + spy_link[8:] + "/build-log.txt"
+    try:
+        build_log_response = requests.get(build_log_url, verify=False, timeout=15)
+        if 'ppc64le' in spy_link:      
+            if job_platform == "libvirt":
+                job_platform+="-ppc64le"
+            elif job_platform == "powervs":
+                job_platform+="-[1-9]"
+            lease = get_lease(build_log_response,job_platform)
+            nightly = get_nightly(build_log_url,build_log_response, 'ppc64le')
+
+        elif 's390x' in spy_link:     
+            if job_platform == "libvirt":
```

Comment:
s390x jobs only run on libvirt, so I think the if condition is not required here.

Reply:
Done.